contentcuration/contentcuration/viewsets/invitation.py

```diff
@@ -2,50 +2,72 @@
 from django_filters.rest_framework import CharFilter
 from django_filters.rest_framework import DjangoFilterBackend
 from django_filters.rest_framework import FilterSet
+from rest_framework import serializers
 from rest_framework.response import Response
 from rest_framework.status import HTTP_204_NO_CONTENT
-from rest_framework import serializers
-from rest_framework_bulk import BulkSerializerMixin
 
 from contentcuration.models import Invitation
-from contentcuration.models import User
+from contentcuration.viewsets.base import BulkListSerializer
+from contentcuration.viewsets.base import BulkModelSerializer
 from contentcuration.viewsets.base import ValuesViewset
+from contentcuration.viewsets.sync.constants import INVITATION
+from contentcuration.viewsets.sync.utils import add_event_for_user
+from contentcuration.viewsets.sync.utils import generate_update_event
 
 
-class InvitationSerializer(BulkSerializerMixin, serializers.ModelSerializer):
+class InvitationSerializer(BulkModelSerializer):
     accepted = serializers.BooleanField(default=False)
+    declined = serializers.BooleanField(default=False)
 
     class Meta:
         model = Invitation
-        fields = ("id", "accepted", "email", "channel", "share_mode")
-        read_only_fields = ("id",)
-
-    def save(self, **kwargs):
-        created = self.instance is None
-        if "email" in self.validated_data:
-            try:
-                self.validated_data["invited"] = User.objects.get(
-                    email=self.validated_data["email"]
-                )
-                self.validated_data.pop("email")
-            except User.DoesNotExist:
-                pass
-        if created and "request" in self.context:
+        fields = ("id", "accepted", "declined", "email", "channel", "share_mode", "first_name", "last_name")
+        list_serializer_class = BulkListSerializer
+
+    def create(self, validated_data):
```

Comment:
This is to make sure someone doesn't try to call the endpoint directly and invite people to the channel that shouldn't have permissions. Invitation.accept is specifically for the invited user to accept, so it would be too late to prevent this action. Since we don't yet allow removing editors, there won't be a case where the sender doesn't have permissions any more